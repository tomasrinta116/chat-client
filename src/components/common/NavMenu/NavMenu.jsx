import React from "react";
import IconButton from "@material-ui/core/IconButton";
import Menu from "@material-ui/core/Menu";
import MenuIcon from "@material-ui/icons/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import { Link } from "react-router-dom";
import { getChannel, exitChannel } from "../../../utils/sendbirdHelpers";

class NavMenu extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      anchorEl: null
    };
  }

  handleClick = event => {
    this.setState({
      anchorEl: event.currentTarget
    });
  };

  handleClose = event => {
    this.setState({ anchorEl: null });
  };

  handleLogout = async () => {
    const { sb, match } = this.props;
    if (match) {
      const channelURL = match.params.channelURL;
      let channel = await getChannel(sb, channelURL);
      await exitChannel(channel);
    }
    sb.disconnect();
  };

  render() {
    const { anchorEl } = this.state;
    return (
      <React.Fragment>
        <IconButton onClick={this.handleClick} color="inherit">
          <MenuIcon />
        </IconButton>
        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
        >
          <MenuItem onClick={this.handleLogout} component={Link} to={"/login"}>
            Logout
          </MenuItem>
        </Menu>
      </React.Fragment>
    );
  }
}

export default NavMenu;
