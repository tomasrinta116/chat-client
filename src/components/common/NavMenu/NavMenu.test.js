import React from "react";
import { shallow } from "enzyme";
import NavMenu from "./NavMenu";

describe("<NavMenu />", () => {
  test("it renders correctly", () => {
    let wrapper = shallow(<NavMenu />);
    expect(wrapper).toMatchSnapshot();
  });
});
