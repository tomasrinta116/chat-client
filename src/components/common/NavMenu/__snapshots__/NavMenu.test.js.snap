// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavMenu /> it renders correctly 1`] = `
<Fragment>
  <WithStyles(IconButton)
    color="inherit"
    onClick={[Function]}
  >
    <pure(MenuIcon) />
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    anchorEl={null}
    id="simple-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(MenuItem)
      component={[Function]}
      onClick={[Function]}
      to="/login"
    >
      Logout
    </WithStyles(MenuItem)>
  </WithStyles(Menu)>
</Fragment>
`;
